test/TensorFlow/tf_graph_func_decls.swift

```diff
@@ -0,0 +1,17 @@
+// RUN: %target-swift-frontend -Xllvm -tf-dump-intermediates -O -emit-sil %s -verify -verify-ignore-unknown | %FileCheck %s
+
+import TensorFlow
+
+@inline(never)
+func hostCode(_ x: Tensor<Float>) -> Tensor<Float> {
+  return x.toAccelerator()
+}
+
+@TensorFlowGraph
+public func send(_ x: Tensor<Float>) -> Tensor<Float> {
+  let y = x + x
+  // TODO(rxwei): Expected error but unknown location "host code is not allowed
+  // in a @convention(tensorflow) function"
```

Comment:
what is the relationship between @convention(tensorflow) and @TensorFlowGraph? from the PR description and change in lib/SIL/SILFunctionType.cpp, it seems we expect them to be used in conjunction in some way. Can you elaborate on this design (i assume @convention(tensorflow) by itself has some limitations as Chris alluded to at the comments), and add some examples or unit tests to illustrate how they should be used together?

Reply:
`@convention(tensorflow)`, like `convention(c)`, is a type attr, not a decl attr. This is not applicable to function declarations. Similarly, to create a C-convention function, you'll need a closure syntax. This is the limitation of the `@convention` attribute.

```swift
@convention(c) // Not okay. A type attribute cannot be used on a decl.
func somefunc(_ x: T) -> T { body }
```
```swift
let somefunc: @convention(c) (T) -> T = { x in  // Okay. Type attribute used on a type
  // body
}
```

Same for `@convention(tensorflow)`.

Since restricting the creation of TensorFlow functions to the closure syntax is not natural, we need a *decl attribute* to imply `@convention(tensorflow)`. Therefore, whenever a function decl has `@TensorFlowGraph`, it implies that the function has a `@convention(tensorflow)` type.